src/olympia/blocklist/admin.py

```diff
@@ -141,14 +129,75 @@ class MultiBlockSubmitAdmin(admin.ModelAdmin):
     )
     ordering = ['-created']
     view_on_site = False
-    list_select_related = ('updated_by',)
+    list_select_related = ('updated_by', 'signoff_by')
+    change_form_template = 'blocklist/block_submission_change_form.html'
+
+    class Media:
+        css = {
+            'all': ('css/admin/blocklist_blocksubmission.css',)
+        }
 
     def has_delete_permission(self, request, obj=None):
+        # For now, keep all BlockSubmission records.
+        # TODO: define under what cirumstances records can be safely deleted.
         return False
 
-    # Read-only mode
     def has_change_permission(self, request, obj=None):
-        return False
+        has = super().has_change_permission(request, obj=obj)
+        pending = obj and obj.signoff_state == BlockSubmission.SIGNOFF_PENDING
+        return has and (not obj or pending)
+
+    def has_signoff_permission(self, request, obj=None):
+        has = self.has_change_permission(request, obj=obj)
+        pending = obj and obj.signoff_state == BlockSubmission.SIGNOFF_PENDING
+        return has and obj and pending and obj.can_user_signoff(request.user)
```

Comment:
The two functions above would benefit from having comments to explain the logic behind these conditions.

Reply:
https://github.com/mozilla/addons-server/pull/13266/commits/61755adf93508dca052d6613741371266942c175